samgeo/samgeo3.py

```
@@ -4148,6 +4160,42 @@ def __init__(
+        This method shuts down the previously tracked predictor (which notifies
+        its workers and destroys its process group) and, as a fallback, destroys
+        any process group that is still initialized.
+        """
+        import torch
+
+        previous = SamGeo3Video._active_predictor
+        if previous is not None:
+            try:
+                previous.shutdown()
+            except Exception:
+                pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)